lib/_http_client.js

```diff
@@ -106,52 +105,53 @@ function ClientRequest(options, cb) {
       method === 'DELETE' ||
       method === 'OPTIONS' ||
       method === 'CONNECT') {
-    self.useChunkedEncodingByDefault = false;
+    this.useChunkedEncodingByDefault = false;
   } else {
-    self.useChunkedEncodingByDefault = true;
+    this.useChunkedEncodingByDefault = true;
   }
 
   if (Array.isArray(options.headers)) {
-    self._storeHeader(self.method + ' ' + self.path + ' HTTP/1.1\r\n',
+    this._storeHeader(this.method + ' ' + this.path + ' HTTP/1.1\r\n',
                       options.headers);
-  } else if (self.getHeader('expect')) {
-    self._storeHeader(self.method + ' ' + self.path + ' HTTP/1.1\r\n',
-                      self._renderHeaders());
+  } else if (this.getHeader('expect')) {
+    this._storeHeader(this.method + ' ' + this.path + ' HTTP/1.1\r\n',
+                      this._renderHeaders());
   }
 
-  if (self.socketPath) {
-    self._last = true;
-    self.shouldKeepAlive = false;
-    var conn = self.agent.createConnection({ path: self.socketPath });
-    self.onSocket(conn);
-  } else if (self.agent) {
+  if (this.socketPath) {
+    this._last = true;
+    this.shouldKeepAlive = false;
+    var conn = this.agent.createConnection({ path: this.socketPath });
+    this.onSocket(conn);
+  } else if (this.agent) {
     // If there is an agent we should default to Connection:keep-alive,
     // but only if the Agent will actually reuse the connection!
     // If it's not a keepAlive agent, and the maxSockets==Infinity, then
     // there's never a case where this socket will actually be reused
-    if (!self.agent.keepAlive && !Number.isFinite(self.agent.maxSockets)) {
-      self._last = true;
-      self.shouldKeepAlive = false;
+    if (!this.agent.keepAlive && !Number.isFinite(this.agent.maxSockets)) {
+      this._last = true;
+      this.shouldKeepAlive = false;
     } else {
-      self._last = false;
-      self.shouldKeepAlive = true;
+      this._last = false;
+      this.shouldKeepAlive = true;
     }
-    self.agent.addRequest(self, options);
+    this.agent.addRequest(this, options);
   } else {
     // No agent, default to Connection:close.
-    self._last = true;
-    self.shouldKeepAlive = false;
+    this._last = true;
+    this.shouldKeepAlive = false;
     if (options.createConnection) {
       var conn = options.createConnection(options);
     } else {
       debug('CLIENT use net.createConnection', options);
       var conn = net.createConnection(options);
     }
-    self.onSocket(conn);
+    this.onSocket(conn);
   }
 
-  self._deferToConnect(null, null, function() {
-    self._flush();
+  var self = this;
+  this._deferToConnect(null, null, () => {
+    this._flush();
     self = null;
```

Comment:
This was rather interesting as well.

`this = null` will throw an error, but a pointer to `this` being assigned `null` will not (obviously).

The concern is that everything above this line was referencing `self`, so if `self` is purposely set to `null` to essentially remove all of that code, replace `self` with `this` would break, but all tests passed.

That is if my assumptions are correct?